src/mscorlib/shared/System/Span.cs

```diff
@@ -508,12 +507,9 @@ public static ReadOnlySpan<char> AsSpan(this string text)
             if (text == null)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.text);
 
-            return new ReadOnlySpan<char>(ref text.GetFirstCharRef(), text.Length);
+            return new ReadOnlySpan<char>(ref text.GetRawStringData(), text.Length);
         }
-    }
 
-    internal static class SpanHelper
```

Comment:
Should we separate the Span/SpanHelper class into a separate file, esp since it has internal statics used by both Span and ReadOnlySpan?